app/code/Magento/MediaGallery/Model/ResourceModel/DeleteAssetsByPaths.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGallery\Model\ResourceModel;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Framework\Exception\CouldNotDeleteException;
+use Magento\MediaGalleryApi\Api\DeleteAssetsByPathsInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Remove asset(s) that correspond the provided path
+ */
+class DeleteAssetsByPaths implements DeleteAssetsByPathsInterface
+{
+    private const TABLE_MEDIA_GALLERY_ASSET = 'media_gallery_asset';
+    private const MEDIA_GALLERY_ASSET_PATH = 'path';
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * DeleteById constructor.
+     *
+     * @param ResourceConnection $resourceConnection
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        LoggerInterface $logger
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(array $paths): void
+    {
+        $failedPaths = [];
+
+        foreach ($paths as $path) {
+            try {
+                $this->validateDirectoryPath($path);
```

Comment:
Move validation step to separate class in model namespace. And create an interface for it in model namespace

Reply:
That is a check if the path is empty. I don't think extracting the class/interface for this purpose is necessary.